forc-plugins/forc-client/src/op/call.rs

```diff
@@ -0,0 +1,2363 @@
+use crate::{
+    cmd,
+    constants::DEFAULT_PRIVATE_KEY,
+    util::{
+        node_url::{get_explorer_url, get_node_url},
+        tx::{prompt_forc_wallet_password, select_local_wallet_account},
+    },
+};
+use anyhow::{anyhow, bail, Result};
+use either::Either;
+use fuel_abi_types::abi::unified_program::UnifiedProgramABI;
+use fuel_tx::{PanicReason, Receipt};
+use fuels::{
+    crypto::SecretKey,
+    programs::calls::{
+        receipt_parser::ReceiptParser,
+        traits::{ContractDependencyConfigurator, TransactionTuner},
+        ContractCall,
+    },
+};
+use fuels_accounts::{provider::Provider, wallet::WalletUnlocked};
+use fuels_core::{
+    codec::{encode_fn_selector, ABIDecoder, ABIEncoder, DecoderConfig, EncoderConfig},
+    types::{
+        bech32::Bech32ContractId,
+        param_types::ParamType,
+        transaction::{Transaction, TxPolicies},
+        transaction_builders::{BuildableTransaction, ScriptBuildStrategy, VariableOutputPolicy},
+        ContractId, EnumSelector, StaticStringToken, Token, U256,
+    },
+};
+use std::{collections::HashMap, fmt::Write, fs::File, io::Write as _, str::FromStr};
+use sway_ast::Item;
+use swayfmt::parse::with_handler;
+
+/// A command for calling a contract function.
+pub async fn call(cmd: cmd::Call) -> anyhow::Result<String> {
+    let cmd::Call {
+        contract_id,
+        abi,
+        function,
+        args,
+        node,
+        caller,
+        call_parameters,
+        mode,
+        gas,
+        external_contracts,
+    } = cmd;
+    let node_url = get_node_url(&node, &None)?;
+    let provider: Provider = Provider::connect(node_url.clone()).await?;
+
+    let wallet = get_wallet(caller, provider).await?;
+
+    if let Some(abi) = abi {
+        // If ABI is provided, ensure function signature is just the selector
+        let cmd::call::FuncType::Selector(selector) = function else {
+            bail!("Function must be a selector");
```

Comment:
It would be good to provide an example here:
```suggestion
            bail!("Function must be a selector, e.g \"transfer\"");
```

Even better would be if we could try to correctly print the function name that they entered.

Reply:
https://github.com/FuelLabs/sway/pull/6791/commits/931b825065a49a75bae67e56bbd61138e8bc992d